tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp

```diff
@@ -848,19 +849,31 @@ std::string SwiftLangSupport::resolvePathSymlinks(StringRef FilePath) {
 
   return InputPath;
 #else
-  char full_path[MAX_PATH];
-
-  HANDLE fileHandle = CreateFileA(
-      InputPath.c_str(), GENERIC_READ, 0, nullptr, OPEN_EXISTING,
-      FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_OPEN_REPARSE_POINT, nullptr);
+  wchar_t full_path[MAX_PATH] = {0};
+  llvm::SmallVector<llvm::UTF16, 50> utf16Path;
+  llvm::convertUTF8ToUTF16String(InputPath.c_str(), utf16Path);
+
+  HANDLE fileHandle = CreateFileW(
+      (LPCWSTR)utf16Path.data(), 
+      0, 
+      FILE_SHARE_READ | FILE_SHARE_WRITE, 
+      nullptr, 
+      OPEN_EXISTING,
+      FILE_FLAG_BACKUP_SEMANTICS,
```

Comment:
Why the backup semantics?

Reply:
It's required to obtain a directory to a handle.